jormungandr/src/blockchain/candidate.rs

```diff
@@ -393,6 +434,9 @@ impl CandidateForest {
                         debug_assert!(block.header().hash() == block_hash);
                         Ok(())
                     }
+                    CandidateData::Applied(_) => {
+                        panic!("caching block {} which is already in storage", block_hash)
```

Comment:
aren't you worried this will blow in production in some cases? can't we just "accept" the state and continue? If the block is already applied then this is not necessarily cause for panic, right?

Reply:
This should never happen as the blocks are checked with storage first before being cached.